tests/microos/install_image.pm

```diff
@@ -0,0 +1,96 @@
+
+    my $device_layout = script_output("lsblk");
+    record_info("Device layout", ${device_layout});
+
+    # Modify disk to be able to correctly boot and login
+    assert_script_run("mount ${root_partition} /mnt");
+    assert_script_run("btrfs property set /mnt ro false");
+    # Set correct serial console to be able to see login in first boot
+    assert_script_run("sed -i 's/console=ttyS0,115200/console=ttyS1,115200/g' /mnt/boot/grub2/grub.cfg") if is_x86_64;
+    # Upload original grub configuration
+    upload_logs("/mnt/etc/default/grub", failok => 1);
+    # Set permanent grub configuration
+    assert_script_run("sed -i 's/console=ttyS0,115200/console=ttyS1,115200/g' /mnt/etc/default/grub") if is_x86_64;
+    # Fully disable graphical terminal on legacy systems without UEFI
+    assert_script_run("sed -i 's/#GRUB_TERMINAL=console/GRUB_TERMINAL=console/g' /mnt/etc/default/grub") if (is_ipmi && !get_var('IPXE_UEFI'));
+    # We need to properly set grub terminal bsc#1230844, otherwise grub menu will not be visible in non graphical boot environments
+    assert_script_run("sed -i 's/GRUB_TERMINAL_INPUT=\".*\"/GRUB_TERMINAL_INPUT=\"console gfxterm\"/g' /mnt/etc/default/grub");
+    assert_script_run("sed -i 's/GRUB_TERMINAL_OUTPUT=\".*\"/GRUB_TERMINAL_OUTPUT=\"console gfxterm\"/g' /mnt/etc/default/grub");
+    # Enable root loging with password
+    assert_script_run("echo 'PermitRootLogin yes' > /mnt/etc/ssh/sshd_config.d/root.conf");
+    assert_script_run("btrfs property set /mnt ro true");
+    assert_script_run("umount /mnt");
+
+    # Setup ignition parition on the end of the same disk and resize root partition to use all the space
+    assert_script_run("printf \"fix\n\" | parted ---pretend-input-tty ${device} print");
+    assert_script_run("parted ${device} --script mkpart primary ext4 98% 100%");
+    assert_script_run("mkfs.ext4 -F ${ignition_partition}");
+    assert_script_run("e2label ${ignition_partition} ignition");
+    assert_script_run("mount ${ignition_partition} /mnt/");
+    assert_script_run("mkdir /mnt/ignition");
+    assert_script_run("curl -v -o /mnt/ignition/config.ign " . data_url("microos/ignition/config.ign"));
+    assert_script_run('umount /mnt');
+
+    # Resize root filesystem to maximum size to use all space up to partition with ignition
+    assert_script_run("parted ${device} --script resize ${root_partition_id} 98%");
+    assert_script_run("mount ${root_partition} /mnt");
+    assert_script_run("btrfs filesystem resize max /mnt");
+    assert_script_run("umount /mnt");
+
+    # We have to use force option to reboot command as installer doesn't have fully running systemd environment
+    power_action("reboot", textmode => 1, force => 1);
+
+    # We can't use reconnect_mgmt_console as it expects fully configured grub, which we don't have at this stage yet
+    select_console "sol", await_console => 0 if is_ipmi;
+    select_console "novalink-ssh", await_console => 0 if is_spvm;
```

Comment:
Not supported, it works partially, we just have old cpu on workers. Not sure if to keep it here or not.

Reply:
I would remove spvm, and simply die on it. We can discuss with IBM its worthyness later

Reply:
Also we plan to eventually move all Novalink instances to the HMC backend as well, see https://progress.opensuse.org/issues/157777

Reply:
Let's remove it, novalink is obsolete anyway and poo#157777 makes sense.